src/main/java/org/ebml/EBMLReader.java

```diff
@@ -279,7 +279,12 @@ public static long readEBMLCode(final ByteBuffer source)
     if (numBytes > 1)
     {
       // Read the rest of the size.
-      data.put(source);
+      //data.put(source);
+      // bug java.nio.BufferOverflowException
+      for (int i = 1; i < numBytes; i++)
+      {
+        data.put(source.get());
+      }
     }
```

Comment:
Well, if you use a loop anyway, you can simply remove the outer `if (numBytes > 1)`. The following should suffice:

```java
    // Read the rest of the size.
    for (int i = 1; i < numBytes; i++)
    {
      data.put(source.get());
    }
```